src/config.rs

```diff
@@ -223,5 +250,53 @@ mod test {
         clean_up(&test_root);
     }
 
+    #[test]
+    fn can_get_inhabitable_shells() {
+        let test_root = set_up("can_get_inhabitable_shells",
+                               "default",
+                               vec!["default", "bcd", "abc", "cde"]);
+        let config = FsConfig::new(test_root.clone());
+        let res = config.get_shell_list().unwrap();
+
+        assert_eq!(res[0], "abc");
+        assert_eq!(res[1], "bcd");
+        assert_eq!(res[2], "cde");
+        assert_eq!(res[3], "default");
+        clean_up(&test_root);
+    }
+
+    #[test]
+    fn can_get_zero_inhabitable_shells() {
+        let test_root = set_up("can_get_zero_inhabitable-shells", "", Vec::new());
+        let config = FsConfig::new(test_root.clone());
+        let res = config.get_shell_list();
+
+        assert_eq!(res.unwrap().len(), 0);
+        clean_up(&test_root);
+    }
+
+    #[test]
+    fn cant_get_inhabitable_shells_for_nonexistant_shell_root() {
+        let config = FsConfig::new(PathBuf::from("not_a_path"));
+        let res = config.get_shell_list();
+        assert!(res.is_err());
+    }
 
+    #[test]
+    fn can_ignore_in_inhabitable_shells_non_unicode_char() {
+        let test_root = set_up("can_ignore_bad_characters",
+                               "default",
+                               vec!["default", "bcd", "abc", "cde"]);
+        let non_unicode = OsString::from_vec((vec![245, 246, 247, 245]));
+        let shell_root = test_root.join("shells").join(non_unicode);
+        fs::create_dir(&shell_root).unwrap();
+        let config = FsConfig::new(test_root.clone());
+        let res = config.get_shell_list().unwrap();
+        assert_eq!(res.len(), 4);
+        assert_eq!(res[0], "abc");
+        assert_eq!(res[1], "bcd");
+        assert_eq!(res[2], "cde");
+        assert_eq!(res[3], "default");
```

Comment:
I wonder if we could make this a bit more concise by saying `assert_eq!(res, vec!["abc", "bcd", "cde", "default"[);`